Travis failures in "devfile catalog, watch command integration tests on kubernetes cluster"

I observed following Travis flake in **devfile catalog, watch command integration tests on kubernetes cluster** test.
```
Created dir: /tmp/483916658
Creating a new project: bmlfjlszob
Running kubectl with args [kubectl create namespace bmlfjlszob]
[kubectl] namespace "bmlfjlszob" created
Running kubectl with args [kubectl config set-context /tmp/483916658 --namespace bmlfjlszob]
[kubectl] error: open /home/travis/.kube/config.lock: file exists
Deleting project: 
Running kubectl with args [kubectl delete namespaces ]
[kubectl] error: resource name may not be empty
• Failure in Spec Setup (BeforeEach) [1.766 seconds]
odo devfile watch command tests
/home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_watch_test.go:14
  when executing watch without a valid devfile [BeforeEach]
  /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_watch_test.go:71
    should fail
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_watch_test.go:72
    No future change is possible.  Bailing out early after 0.626s.
    Running kubectl with args [kubectl config set-context /tmp/483916658 --namespace bmlfjlszob]
    Expected
        <int>: 1
    to match exit code:
        <int>: 0
    /home/travis/gopath/src/github.com/openshift/odo/tests/helper/helper_run.go:34
------------------------------
SSS•••
Summarizing 1 Failure:
[Fail] odo devfile watch command tests [BeforeEach] when executing watch without a valid devfile should fail 
/home/travis/gopath/src/github.com/openshift/odo/tests/helper/helper_run.go:34
Ran 4 of 35 Specs in 3.813 seconds
FAIL! -- 3 Passed | 1 Failed | 0 Pending | 31 Skipped

```
/kind flake
_Originally posted by @dharmit in https://github.com/openshift/odo/pull/2944#issuecomment-617552849_

https://github.com/openshift/odo/pull/2944#issuecomment-617746856

Saw this recently on Travis. 
```
$ make goget-tools
go get -u github.com/Masterminds/glide
go get -u github.com/frapposelli/wwhrd
go get -u github.com/onsi/ginkgo/ginkgo
go get -u github.com/securego/gosec/cmd/gosec
# It is not recomended to go get golangci-lint https://github.com/golangci/golangci-lint#go
curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b /home/travis/gopath/bin v1.23.6
golangci/golangci-lint info checking GitHub for tag 'v1.23.6'
golangci/golangci-lint info found version: 1.23.6 for v1.23.6/linux/amd64
make: *** [goget-tools] Error 7
The command "make goget-tools" failed and exited with 2 during .
```

--- a/tests/helper/kubernetes_utils.go
+++ b/tests/helper/kubernetes_utils.go
@@ -2,10 +2,19 @@ package helper
 
 import (
 	"fmt"
+	"os"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
+
+// CopyKubeConfigFile copies default kubeconfig file into current context config file
+func CopyKubeConfigFile(kubeConfigFile, tempConfigFile string, info os.FileInfo) string {
+	err := copyFile(kubeConfigFile, tempConfigFile, info)
+	Expect(err).NotTo(HaveOccurred())
+	os.Setenv("KUBECONFIG", tempConfigFile)
+	return tempConfigFile
+}
 
 // CreateRandNamespace create new project with random name in kubernetes cluster (10 letters)
 func CreateRandNamespace(context string) string {

--- a/tests/integration/devfile/cmd_devfile_catalog_test.go
+++ b/tests/integration/devfile/cmd_devfile_catalog_test.go
@@ -22,7 +22,10 @@ var _ = Describe("odo devfile catalog command tests", func() {
 		os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
 		helper.CmdShouldPass("odo", "preference", "set", "Experimental", "true")
 		if os.Getenv("KUBERNETES") == "true" {
-			project = helper.CreateRandNamespace(context)
+			info, err := os.Stat(os.Getenv("KUBECONFIG"))
+			Expect(err).NotTo(HaveOccurred())
+			kubeConfigFile := helper.CopyKubeConfigFile(os.Getenv("KUBECONFIG"), filepath.Join(context, "config"), info)
+			project = helper.CreateRandNamespace(kubeConfigFile)
 		} else {
 			project = helper.CreateRandProject()
 		}
@@ -34,11 +37,13 @@ var _ = Describe("odo devfile catalog command tests", func() {
 	var _ = AfterEach(func() {
 		if os.Getenv("KUBERNETES") == "true" {
 			helper.DeleteNamespace(project)
+			os.Unsetenv("KUBECONFIG")
 		} else {
 			helper.DeleteProject(project)
 		}
 		helper.Chdir(currentWorkingDirectory)
 		helper.DeleteDir(context)
+		os.Unsetenv("GLOBALODOCONFIG")
 	})
 
 	Context("When executing catalog list components", func() {

--- a/tests/integration/devfile/cmd_devfile_watch_test.go
+++ b/tests/integration/devfile/cmd_devfile_watch_test.go
@@ -24,7 +24,10 @@ var _ = Describe("odo devfile watch command tests", func() {
 		context = helper.CreateNewContext()
 		os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
 		if os.Getenv("KUBERNETES") == "true" {
-			namespace = helper.CreateRandNamespace(context)
+			info, err := os.Stat(os.Getenv("KUBECONFIG"))
+			Expect(err).NotTo(HaveOccurred())
+			kubeConfigFile := helper.CopyKubeConfigFile(os.Getenv("KUBECONFIG"), filepath.Join(context, "config"), info)
+			namespace = helper.CreateRandNamespace(kubeConfigFile)
 		} else {
 			namespace = helper.CreateRandProject()
 		}
@@ -37,6 +40,7 @@ var _ = Describe("odo devfile watch command tests", func() {
 	var _ = AfterEach(func() {
 		if os.Getenv("KUBERNETES") == "true" {
 			helper.DeleteNamespace(namespace)
+			os.Unsetenv("KUBECONFIG")
 		} else {
 			helper.DeleteProject(namespace)
 		}

